docs/enhancements/boot-status-reporter.md

```diff
@@ -0,0 +1,129 @@
+   users about the host progress stages (first boot, second boot, etc).
+
+
+2. The new service should be thoroughly cleaned up from the hosts and should never fail installations by itself.
+   If the boot reporter service fails to run or report logs, the cluster installation should continue, 
+   and the new host progress stages should be skipped in such a scenario.
+
+## Proposal
+
+
+### Changes To Assisted Service
+For cases as mentioned in `Motivation`, we need a new service (start via `systemd`) soon as possible after hosts
+are booting from the disk and contact `assisted-service` to:
+
+1. Change the installation stage to a new `Booted with local ignition` Between `Configuring`
+   and  `Waiting for control plane`. This stage refers to a boot with pointer ignition.
+
+
+2. Change the installation stage to a new `Booted with control plane ignition` Between
+   `Waiting for control plane` and  `Joined`.
+
+
+3. For each of the above-mentioned, Read the current host stage from `assisted-service`, to determine if:
+   - the service runs on the first boot :arrow_right: collect logs and report.
+   - or the second boot :arrow_right: collect logs, report, and self-cleanup after a configurable threshold.
+
+### Information Needed And Changes To The Discovery Agent
+
+In order to communicate with `assisted-service`, The assisted boot reported service needs to know:
+1. The `assisted-service` IP address and port.
+2. The pull-secret to use for agent authentication.
+3. The InfraEnv ID and host ID
+
+It will need this information starting from the first boot, which is a pointer ignition-based boot.
+
+For that, we need to modify the discovery agent to store this information on disk in a pre-defined location for the assisted boot reporter service to use.
+
+The assisted boot reporter service should start via `systemd` and determine its operating stage. As mentioned above:
+
+### First Boot (Pointer Ignition)
+Expected stage to be read from assisted-service backend: `Configuring`.
+Update the host stage to: `Booted with local ignition`.
+Start a timer to compress and send logs every 10 minutes to `v2UpdateHostInstallProgress` API endpoint.
+
+
+### Second Boot (Control Plane Ignition)
+Expected stage to be read from assisted-service backend: `Waiting for control plane`.
+Update the host stage to: `Booted with control plane ignition`.
+Start a timer to compress send logs every 10 minutes to `v2UpdateHostInstallProgress` API endpoint.
+Start another timer to halt operation and self-cleanup as follows:
+
+1. If 1 hour has passed.
+2. If cluster install progress reached a point where it is no longer needed (now counting in `assisted-controller` pod). An indication could be that `kubelet` is up and running.
+   Cleanup should include:
+
+   * The stored files for pull-secret, host id, and infraEnv id.
+   * Service itself - should not run on additional host boot.
```

Comment:
@avishayt @tsorya @nmagnezi,
Can someone please explain why we need the boot-reporter service needs to run at this stage (for multi-node cluster installation)?
Perhaps the name (boot-reporter service) is misleading since it seems that it's expected to do more than just report that the host booted...
And it's unclear in which cases it provides the information we don't get without it.

Reply:
For the bootstrap host post reboot issue diagnosting, must gather / controller don't give you that info

Reply:
We should run masters-gather and upload the result